third-party/java/d8/changes.mbox

```diff
@@ -1,24 +1,53 @@
-From fa0678bd211c878ebd82a3dac5534ad1ba723e28 Mon Sep 17 00:00:00 2001
+From 51a9e6acd760fa917f485aec9a36a0d2545a6a29 Mon Sep 17 00:00:00 2001
 From: Gautam Korlam <gkorlam@uber.com>
-Date: Tue, 4 Dec 2018 17:21:07 -0800
-Subject: [PATCH] Compute referenced resources in D8
+Date: Tue, 19 Feb 2019 08:05:45 -0800
+Subject: [PATCH] Changes to D8 to make it work with facebook's buck build
+ system
 
-- Computing referenced resources is used to trim the Uber R.java.
-- Set default min sdk for D8 to Api level J
-- Fix Lambda Descriptor name handling to be compatible with buck's predexing
 ---
- src/main/java/com/android/tools/r8/D8Command.java  |  8 +++++++-
- .../com/android/tools/r8/graph/DexItemFactory.java | 14 ++++++++++++++
- .../tools/r8/ir/conversion/IRConverter.java        |  2 +-
- .../tools/r8/ir/desugar/LambdaDescriptor.java      | 11 +++--------
- .../android/tools/r8/utils/AndroidApiLevel.java    |  3 ++-
- 5 files changed, 27 insertions(+), 11 deletions(-)
+ src/main/java/com/android/tools/r8/D8.java    |  2 +-
+ .../java/com/android/tools/r8/D8Command.java  | 14 +++++++++-
+ .../com/android/tools/r8/graph/AppInfo.java   | 16 ++++++++++-
+ .../tools/r8/graph/DexItemFactory.java        | 14 ++++++++++
+ .../tools/r8/ir/conversion/IRConverter.java   |  2 +-
+ .../r8/ir/desugar/Java8MethodRewriter.java    | 10 +++----
+ .../tools/r8/ir/desugar/LambdaDescriptor.java |  2 +-
+ .../tools/r8/utils/AndroidApiLevel.java       |  3 +-
+ .../android/tools/r8/utils/AndroidApp.java    | 28 +++++++++++++++++--
+ tools/gradle.py                               |  1 +
+ 10 files changed, 78 insertions(+), 14 deletions(-)
 
+diff --git a/src/main/java/com/android/tools/r8/D8.java b/src/main/java/com/android/tools/r8/D8.java
+index 2273c4d4b..5397de771 100644
+--- a/src/main/java/com/android/tools/r8/D8.java
++++ b/src/main/java/com/android/tools/r8/D8.java
+@@ -175,7 +175,7 @@ public final class D8 {
+       options.outline.enabled = false;
+ 
+       DexApplication app = new ApplicationReader(inputApp, options, timing).read(executor);
+-      AppInfo appInfo = new AppInfo(app);
++      AppInfo appInfo = new AppInfo(app, inputApp.getBucketId());
+       app = optimize(app, appInfo, options, timing, executor);
+ 
+       // Close any internal archive providers now the application is fully processed.
 diff --git a/src/main/java/com/android/tools/r8/D8Command.java b/src/main/java/com/android/tools/r8/D8Command.java
-index 10045324b..63dc33ac0 100644
+index 10045324b..580d0b2dc 100644
 --- a/src/main/java/com/android/tools/r8/D8Command.java
 +++ b/src/main/java/com/android/tools/r8/D8Command.java
-@@ -176,6 +176,11 @@ public final class D8Command extends BaseCompilerCommand {
+@@ -105,6 +105,12 @@ public final class D8Command extends BaseCompilerCommand {
+       return self();
+     }
+ 
++    /**  Set unique identifier of a bucket of Java classes that D8 will dex independently in parallel. */
```

Comment:
This enables d8 to have a unique identifier per dex bucket i.e classes2.dex, classes3.dex etc. so that thesynthetic classnames are not duplicated and end up conflicting in redex.

The change to buck to actually pass in this id will be opened in a separate PR.

cc @awelc